File: 배열값 중 0값을 찾아 배열 뒤로 배치시키기/sol.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    vector<int> nums = {6, 0, 8, 2, 3, 0, 4, 0, 1};
    
    for (int left = 0, right = 0; right < nums.size(); ++right) {
        if (nums[right]) {
            swap(nums[left++], nums[right]);
        }
    }

    for (int num : nums) {
        cout << num << " ";
    }
    
    return 0;
}

File: 배열값 중 0값을 찾아 배열 뒤로 배치시키기/sol.java
import java.io.*;
import java.util.Arrays;

class Main {
    
    public static void swap(int[] nums, int left, int right) {
        int temp = nums[left];
        nums[left] = nums[right];
        nums[right] = temp;
    }
    
	public static void main (String[] args) {
	    int[] nums = {6, 0, 8, 2, 3, 0, 4, 0, 1};
	    
	    for (int left = 0, right = 0; right < nums.length; ++right) {
            if (nums[right] != 0) {
                swap(nums, left++, right);
            }
	    }
	    
        for (int num : nums) {
            System.out.print(num + " ");
        }
	}
	
}